test/extensions/tracers/xray/tracer_test.cc

```diff
@@ -183,6 +183,16 @@ TEST_F(XRayTracerTest, TraceIDFormatTest) {
   ASSERT_EQ(24, parts[2].length());
 }
 
+TEST(XRayDaemon, SendBytesOverUDP) {
```

Comment:
This doesn't actually verify anything, right? Can you look at the tests I added for statsd not that long ago? You should be able to get a real working UDP test running quite quickly which will give you real coverage.

Reply:
It doesn't verify anything. I was mainly trying to get it to run through the sanitizers to flush out any memory safety bugs.

I'll look at the statsd tests

Reply:
I added a UDP test. I had to make a small change to the utility class to make it listen on ports other than loopback.